### PR TITLE
[Feature] Hyper Embedding Support(AttH)

### PR DESCRIPTION
**Pull request by menjarleev**
*Issue #, if available:*
* multi-gpu async deployment in test
* the new framework is overall 4 times slower than the general_model

*Description of changes:*
* generalize ke_model structure so that the elder version can easily migrate to it.
* add AttH support. Best hyper-parameter is in search.
* add TransE_l1 for ke_model structure for profile

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/dglke/dataloader/sampler.py [hunk 1]
```diff
@@ -678,6 +688,13 @@ def create_sampler(self, eval_type, batch_size, neg_sample_size, neg_chunk_size,
         return EvalSampler(self.g, edges, batch_size, neg_sample_size, neg_chunk_size,
                            mode, num_workers, filter_false_neg)
 
+    def part_edge(self, rank, world_size, mode):
+        edges = self.get_edges(mode)
+        beg = edges.shape[0] * rank // world_size
+        end = min(edges.shape[0] * (rank + 1) // world_size, edges.shape[0])
```
Comment (classicsong) [inline python/dglke/dataloader/sampler.py:694]: You may loss some tailing edges. Use:
```
step_size = (edges.shape[0] + world_size - 1) // world_size
beg = step_size * rank
end = min(step_size * (rank + 1), edges, shape[0])
```

Comment (menjarleev) [reply python/dglke/dataloader/sampler.py:694]: corrected.

**Comment by zheng-da** [comment]
A general comment: please split the PR into multiple smaller PRs. Each PR should only do one thing. It'll be much easier to review.
